core/Db/TransactionLevel.php

```diff
@@ -40,12 +40,18 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        if ($this->db->supportsUncommitted === false) {
+            // we know "Uncommitted" transaction level is not supported, we don't need to do anything as it won't work to set the status
+            return;
```

Comment:
shouldn't that be to have consistent return types:
```suggestion
            return false;
```